packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1134,25 +1125,27 @@ function finishConcurrentRender(
       throw new Error('Root did not complete. This is a bug in React.');
     }
     case RootSuspendedWithDelay: {
-      if (includesOnlyTransitions(lanes)) {
-        // This is a transition, so we should exit without committing a
-        // placeholder and without scheduling a timeout. Delay indefinitely
-        // until we receive more data.
-        if (enableProfilerTimer && enableComponentPerformanceTrack) {
-          finalizeRender(lanes, now());
-        }
-        const didAttemptEntireTree =
-          !workInProgressRootDidSkipSuspendedSiblings;
-        markRootSuspended(
-          root,
-          lanes,
-          workInProgressDeferredLane,
-          didAttemptEntireTree,
-        );
-        return;
+      if (!includesOnlyTransitions(lanes)) {
```

Comment:
This is the subtle difference that `RootSuspendedWithDelay` checks the lanes but maybe we should just do that before setting the flag in the first place?